File: infermap/scorers/exact.py
"""Exact name scorer — case-insensitive exact field name match."""
from __future__ import annotations

from infermap.types import FieldInfo, ScorerResult


class ExactScorer:
    """Returns 1.0 when source and target field names match exactly (case-insensitive)."""

    name: str = "ExactScorer"
    weight: float = 1.0

    def score(self, source: FieldInfo, target: FieldInfo) -> ScorerResult:
        src = source.name.strip().lower()
        tgt = target.name.strip().lower()
        if src == tgt:
            return ScorerResult(score=1.0, reasoning=f"Exact name match: '{source.name}'")
        return ScorerResult(score=0.0, reasoning=f"No exact match: '{source.name}' vs '{target.name}'")

File: packages/infermap-js/src/core/scorers/exact.ts
// Exact name scorer — case-insensitive exact field name match.
// Mirrors infermap/scorers/exact.py.
import type { FieldInfo, ScorerResult } from "../types.js";
import { makeScorerResult } from "../types.js";
import type { Scorer } from "./base.js";

export class ExactScorer implements Scorer {
  readonly name = "ExactScorer";
  readonly weight = 1.0;

  score(source: FieldInfo, target: FieldInfo): ScorerResult {
    const src = source.name.trim().toLowerCase();
    const tgt = target.name.trim().toLowerCase();
    if (src === tgt) {
      return makeScorerResult(1.0, `Exact name match: '${source.name}'`);
    }
    return makeScorerResult(
      0.0,
      `No exact match: '${source.name}' vs '${target.name}'`
    );
  }
}
